code/game/objects/items/grenades/chem_grenade.dm

```diff
@@ -71,8 +67,11 @@
 				I.play_tool_sound(src, 25)
 			else
 				to_chat(user, "<span class='warning'>You need to add at least one beaker before locking the [initial(name)] assembly!</span>")
-		else if(stage == READY && !nadeassembly && !active)
-			change_det_time()
+		else if(stage == READY)
+			det_time = det_time == 50 ? 30 : 50 //toggle between 30 and 50
+			if(landminemode)
+				landminemode.time = det_time * 0.1	//overwrites the proxy sensor activation timer and puts between 30 and 50

```

Comment:
30 * 0.1 is not in between 30 & 50 unless i'm misreading something

Reply:
proxy.time reads numbers in seconds

Reply:
oh so it just converts to seconds and doesn't do the second half of the comment.